### PR TITLE
Add method to sign raw data, enabling easier device app testing

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Currently, the ledger-solana-app signs data blindly, which means a Ledger can successfully sign random (non-transaction) data. This is useful for testing device and transport configuration (data length limits, etc), but there is no public api available to take advantage of this functionality.

#### Summary of Changes
Add `sign_raw_data` method. This will be deprecated or updated to use a different command number in the future, once on-device message parsing lands.

### CODE REVIEW

File: remote-wallet/src/ledger.rs [hunk 1]
```diff
@@ -233,6 +233,19 @@ impl LedgerWallet {
             ver[3].into(),
         ))
     }
+
+    pub fn sign_raw_data(
+        &self,
+        derivation: &DerivationPath,
+        data: &[u8],
+    ) -> Result<Vec<u8>, RemoteWalletError> {
+        let mut payload = get_derivation_path(&derivation);
```
Comment (garious) [inline remote-wallet/src/ledger.rs]: Even though this is temporary, this is confusing naming! `get_derivation_path` of something called `derivation` of type `DerivationPath`.

Comment (CriesofCarrots) [reply remote-wallet/src/ledger.rs]: Fair, and that part's not temporary. Better?

Comment (garious) [reply remote-wallet/src/ledger.rs]: yes, thanks

**Comment by CriesofCarrots** [comment]
@garious , I'm going to boldly merge this. Let me know if you have any other concerns and I'll fixup.

**Comment by garious** [comment]
Yeah, generally no need to wait on my reviews

**Pull request merged**